project/driver_controls/inc/hazard_light_state.h

```diff
@@ -0,0 +1,7 @@
+#pragma once
```

Comment:
For all of your FSMs: I believe it's clearer to call them `[module]_fsm` instead of `[module]_state`.